src/xdg.rs

```diff
@@ -90,12 +174,26 @@ impl BaseDirectories
     /// As per specification, if an environment variable contains a relative path,
     /// the behavior is the same as if it was not set.
     pub fn new() -> BaseDirectories {
+        unwrap(BaseDirectories::try_new())
+    }
+
+    /// Same as [`new()`](#method.new), but errors are caught
+    /// and propagated to the caller.
+    pub fn try_new() -> Result<BaseDirectories, BaseDirectoriesError> {
```

Comment:
You can break the API; it's prudent to avoid combinatorial proliferation of panicking and non-panicking functions, just like stdlib does. Note how 2.0 is still a prerelease, specifically for this reason--I plan to cut a release just before Cargo starts using it.